Source/Runtime/Conditions/Poke interaction/PokeableProperty.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using UnityEngine;
+using UnityEngine.XR.Interaction.Toolkit;
+using VRBuilder.Core.Properties;
+using VRBuilder.BasicInteraction.Properties;
+using System.Linq;
+using UnityEngine.XR.Interaction.Toolkit.Filtering;
+
+namespace VRBuilder.XRInteraction.Properties
+{ 
+    /// <summary>
+    /// XR implementation of <see cref="IPokeableProperty"/>.
+    /// </summary>
+    [RequireComponent(typeof(XRSimpleInteractable))]
+    [RequireComponent(typeof(XRPokeFilter))]
+    public class PokeableProperty : LockableProperty, IPokeableProperty
+    {
+        public event EventHandler<EventArgs> Poked;
+        public event EventHandler<EventArgs> Unpoked;
+
+        /// <summary>
+        /// Returns true if the GameObject is touched.
+        /// </summary>
+        public virtual bool IsBeingPoked => Interactable != null && Interactable.interactorsSelecting.Any(i => i.transform.root.GetComponentInChildren<UserSceneObject>() != null);
+
+        protected XRSimpleInteractable Interactable
+        {
+            get
+            {
+                if (interactable == false)
+                {
+                    interactable = GetComponent<XRSimpleInteractable>();
+                }
+
+                return interactable;
+            }
+        }
+
+        private XRSimpleInteractable interactable;
+
+        protected override void OnEnable()
+        {
+            base.OnEnable();
+            
+            Interactable.selectEntered.AddListener(HandleXRPoked);
+            Interactable.selectExited.AddListener(HandleXRUnpoked);
+
+            InternalSetLocked(IsLocked);
+        }
+
+        protected override void OnDisable()
+        {
+            base.OnDisable();
+            
+            Interactable.selectEntered.RemoveListener(HandleXRPoked);
+            Interactable.selectExited.RemoveListener(HandleXRUnpoked);
+        }
+        
+        protected void Reset()
+        {
+            // TODO: Anything to do here?
```

Comment:
Maybe set defaults on the interactable, but not really necessary especially if the interactable does not need extra setup.